invoke/scalewayInvoke.js

```diff
@@ -0,0 +1,82 @@
+const BbPromise = require('bluebird');
+const axios = require('axios');
+const util = require('util');
+
+const scalewayApi = require('../shared/api/endpoint');
+const setUpDeployment = require('../shared/setUpDeployment');
+const validate = require('../shared/validate');
+
+class ScalewayInvoke {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options || {};
+    this.provider = this.serverless.getProvider('scaleway');
+    this.provider.initialize(this.serverless, this.options);
+
+    const api = scalewayApi.getApi(this);
+
+    Object.assign(
+      this,
+      validate,
+      setUpDeployment,
+      api,
+    );
+
+    // Interrogate config to see if we're invoking a container or a function
```

Comment:
@RamyChaabane as discussed offline, I'll add a validation step before this, to throw an error if the function name is not mentioned at all in the serverless config file.

Reply:
@RamyChaabane see validation below.